### PR TITLE
Baseline API

### PR DESCRIPTION
**Pull request by jcace**
closes https://github.com/data-preservation-programs/spade/issues/33

### CODE REVIEW

**Comment by jcace** [comment]
To view the rendered swagger docs, copy the full contents of the generated swagger.yaml file ( https://github.com/data-preservation-programs/spade-tenant/blob/8154adafccb8a8ba3167b
Screenshot 2023-10-31 at 3 47 58 PM
2286b74662be9dc3cb8/docs/swagger.yaml ) into https://editor.swagger.io/

File: api/v1/address.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+type Address struct {
+	Address   string `json:"address"`
+	IsSigning bool   `json:"is_signing"` // true - active dealmaking from this address, false - still counts as an associated wallet
+}
+
+// handleSetAddresses godoc
+//	@Summary		Update addresses associated with a tenant
+// 	@Param 		  token header string true "Auth token"
+//  @Param 			addresses body Addresses true "New addresses to add or change is_signing flag of"
```
Comment (stephen-pl) [inline api/v1/address.go:19]: make pretty

File: api/v1/address.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//	@Success		200	{object}	ResponseEnvelope{response=Addresses}
+//	@Router			/addresses [put]
+func (s *apiV1) handleSetAddresses(c echo.Context) error {
+	return c.JSON(http.StatusNotImplemented, map[string]string{})
+}
+
+// handleDeleteAddresses godoc
+//	@Summary		Delete addresses used by a tenant
+// 	@Param 		  token header string true "Auth token"
+//  @Param 			addresses body []string true "addresses to delete"
```
Comment (stephen-pl) [inline api/v1/address.go:30]: make pretty

File: api/v1/address.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+)
+
+type Addresses []Address
+
+type Address struct {
+	Address   string `json:"address"`
+	IsSigning bool   `json:"is_signing"` // true - active dealmaking from this address, false - still counts as an associated wallet
```
Comment (stephen-pl) [inline api/v1/address.go:13]: Is only one address allowed to sign. Does the address have to sign? What about can_sign?

File: api/v1/collection.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+type CreateCollectionResponse struct {
+	CollectionID uint   `json:"collection_id"`
+	Status       string `json:"status"`
+}
+
+// handleCreateCollection godoc
+//	@Summary		Creates a new collection
+// 	@Param 		  token header string true "Auth token"
+//  @Param 			collection body CreateCollectionRequest true "Collection to create"
```
Comment (stephen-pl) [inline api/v1/collection.go:34]: make pretty

File: api/v1/collection.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+type Collection struct {
+	ID                     string                  `json:"id"` // todo: UUIDs
+	Name                   string                  `json:"name"`
+	ReplicationConstraints []ReplicationConstraint `json:"replication_constraints"`
+}
+
+// handleGetCollections godoc
+//	@Summary		Gets info about collections
+// 	@Param 		  token header string true "Auth token"
```
Comment (stephen-pl) [inline api/v1/collection.go:52]: same

File: api/v1/constraint-labels.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ConstraintLabels}
+//	@Router			/constraint-labels [get]
+func (s *apiV1) handleGetConstraintLabels(c echo.Context) error {
+	return c.JSON(http.StatusNotImplemented, map[string]string{})
+}
+
+// TODO: PUT - for V2
+
+/** example response
```
Comment (stephen-pl) [inline api/v1/constraint-labels.go:29]: does swagger not generate these examples for us?

File: api/v1/eligibility-criteria.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	Attribute string `json:"attribute"`
+	Operator  string `json:"operator"`
+
+	Value interface{} `json:"value"` // TODO: type - either []string or string
+}
+
+// handleSetEligibilityCriteria godoc
+//	@Summary		Set sp eligibility criteria
+// 	@Param 		  token header string true "Auth token"
+//  @Param 			elibility_criteria body EligibilityCriteria true "New eligibility criteria to update to"
```
Comment (stephen-pl) [inline api/v1/eligibility-criteria.go:23]: same

**Comment by stephen-pl** [comment]
Overall looks good to me and what we discussed.

File: api/v1/address.go (hunk 3 above)
Comment (jcace) [reply api/v1/address.go:13]: nope, there can be multiple with `is_signing` 

the specifics about how/which one specifically is used is out of scope, handled by MPL50 and/or remote signing network

File: api/v1/constraint-labels.go (hunk 6 above)
Comment (jcace) [reply api/v1/constraint-labels.go:29]: There's a way of specifying it in-line as an [example value](https://github.com/swaggo/swag#example-value-of-struct), but just for readability sake for now I'll keep this one as a comment. It's moreso a reminder for me about how things will be shaped in the db schema

**Comment by xmcai2016** [comment]
Overall lgtm, great work @jcace :)

**Pull request merged**